optaplanner-benchmark/src/main/resources/org/optaplanner/benchmark/impl/report/benchmarkReport.html.ftl

```diff
@@ -26,16 +26,17 @@
 </#macro>
 <#macro addSingleRankingBadge singleBenchmarkResult>
     <#if !singleBenchmarkResult.ranking??>
-    <span class="badge badge-important">F</span>
+    <span class="badge badge-important" data-toggle="tooltip" title="Failed benchmark">F</span>
+    <#elseif singleBenchmarkResult.winner>
+    <span class="badge badge-success">${singleBenchmarkResult.ranking}</span>
     <#else>
-        <#if singleBenchmarkResult.winner>
-        <span class="badge badge-success">${singleBenchmarkResult.ranking}</span>
-        <#else>
-        <span class="badge">${singleBenchmarkResult.ranking}</span>
-        </#if>
-        <#if !singleBenchmarkResult.scoreFeasible>
-        <span class="badge badge-warning">!</span>
-        </#if>
+    <span class="badge">${singleBenchmarkResult.ranking}</span>
+    </#if>
+
+    <#if !singleBenchmarkResult.initialized>
```

Comment:
Shouldn't this also be an elseif?
If the "Failed" badge is shown, there is no point in showing the "Initialized" or "Infeasible" badge.

I am already merging this PR, as it's a big step forward, but do take a look at this issue.